include/bredis/Command.hpp

```diff
@@ -0,0 +1,41 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+//
+#pragma once
+
+#include <boost/utility/string_ref.hpp>
+#include <boost/variant.hpp>
+#include <type_traits>
+#include <vector>
+
+#include "Result.hpp"
+
+namespace bredis {
+
+template <bool...> struct bool_pack;
+template <bool... bs>
+using all_true = std::is_same<bool_pack<bs..., true>, bool_pack<true, bs...>>;
+
+template <class R, class... Ts>
+using are_all_constructible = all_true<std::is_constructible<R, Ts>::value...>;
```

Comment:
This should go in a "detail" namespace, unless you intend for it to be public? See:
https://github.com/vinniefalco/Beast/blob/master/include/beast/core/detail/type_traits.hpp#L45